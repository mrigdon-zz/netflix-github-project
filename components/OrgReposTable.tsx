import React from "react";
import styles from "../styles/OrgReposTable.module.css";
import classNames from "../utils/classNames";
import { HomeContext } from "../utils/homeContext";
import { label, LocaleContext, timestamp } from "../utils/i18n";
import Label from "./Label";
import Repo from "../githubAPI/Repo";
import TableHeader from "./TableHeader";

export type ColumnId = "repo" | "stars" | "forks" | "updated" | "commits";

export default function OrgReposTable({
  repos,
  onClick,
}: {
  repos: Repo[];
  onClick(id: ColumnId): void;
}) {
  const { locale } = React.useContext(LocaleContext);
  const { params } = React.useContext(HomeContext);

  function formattedCount(stars: number) {
    const formatter = new Intl.NumberFormat(locale);
    return formatter.format(stars);
  }

  const headers = [
    { id: "repo", label: label(locale, "repo") },
    { id: "stars", label: label(locale, "stars") },
    { id: "forks", label: label(locale, "forks") },
    { id: "updated", label: label(locale, "lastUpdated") },
    { id: "commits", label: label(locale, "commits"), unsortable: true },
  ].map((header) => ({
    ...header,
    sorted: header.id === params.column,
    ascending: params.ascending,
    onClick,
  }));

  return (
    <table className={styles.table}>
      <thead>
        <tr>
          {headers.map((header) => (
            <TableHeader key={header.id} {...header} />
          ))}
        </tr>
      </thead>

      <tbody>
        {repos.map((repo) => (
          <tr key={repo.id} className={styles.row}>
            <td className={styles.cell}>
              <a href={repo.html_url} target="_blank" rel="noopener">
                {repo.name}
              </a>
            </td>

            <td className={styles.cell}>
              {formattedCount(repo.stargazers_count)}
            </td>

            <td className={styles.cell}>{formattedCount(repo.forks_count)}</td>

            <td className={styles.cell}>
              {timestamp(locale, repo.updated_at)}
            </td>

            <td className={classNames(styles.cell, styles.viewAll)}>
              <a
                href={`/orgs/${params.search}/repos/${repo.name}?locale=${locale}`}
              >
                <Label name="viewAll" />
              </a>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}
