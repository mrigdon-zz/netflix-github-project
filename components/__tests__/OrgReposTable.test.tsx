import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import OrgReposTable from "../OrgReposTable";

describe("OrgReposTable", () => {
  it("renders the repos", () => {
    const renderer = createRenderer();
    renderer.render(
      <OrgReposTable
        repos={[
          {
            id: 1,
            name: "eureka",
            stargazers_count: 42000,
            updated_at: "2021-09-09T15:18:56.813Z",
            html_url: "https://github.com/netflix/eureka",
            forks_count: 2400,
            default_branch: "main",
          },
        ]}
        onClick={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
