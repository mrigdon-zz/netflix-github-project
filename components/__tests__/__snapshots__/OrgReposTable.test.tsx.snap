// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgReposTable renders the repos 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <TableHeader
        ascending={false}
        id="repo"
        label="📦 Repo"
        onClick={[MockFunction]}
        sorted={false}
      />
      <TableHeader
        ascending={false}
        id="stars"
        label="⭐️ Stars"
        onClick={[MockFunction]}
        sorted={true}
      />
      <TableHeader
        ascending={false}
        id="forks"
        label="🍴 Forks"
        onClick={[MockFunction]}
        sorted={false}
      />
      <TableHeader
        ascending={false}
        id="updated"
        label="✍️ Last Updated"
        onClick={[MockFunction]}
        sorted={false}
      />
      <TableHeader
        ascending={false}
        id="commits"
        label="🖋 Commits"
        onClick={[MockFunction]}
        sorted={false}
        unsortable={true}
      />
    </tr>
  </thead>
  <tbody>
    <tr
      className="row"
    >
      <td
        className="cell"
      >
        <a
          href="https://github.com/netflix/eureka"
          rel="noopener"
          target="_blank"
        >
          eureka
        </a>
      </td>
      <td
        className="cell"
      >
        42,000
      </td>
      <td
        className="cell"
      >
        2,400
      </td>
      <td
        className="cell"
      >
        Sep 9, 2021, 8:18 AM
      </td>
      <td
        className="cell viewAll"
      >
        <a
          href="/orgs//repos/eureka?locale=en"
        >
          <Label
            name="viewAll"
          />
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
